File: task3/GO/main.go
package main

import (
	"fmt"
)

func main() {
	var N int
	fmt.Scan(&N)
	if N <= 1 {
		fmt.Println(0)
		return
	}

	var now int
	var last int
	count := 0

	fmt.Scan(&last)

	for i := 0; i < N-1; i++ {
		fmt.Scan(&now)

		if now == last {
			count++
		}
		last = now
	}
	fmt.Println(count)
}

File: task3/CPP/main.cpp
#include <iostream>

using namespace std;

int main() {
    int N;
    cin >> N;
    if (N <= 1) {
        cout << 0 << endl;
        return 0;
    }
 
    int now; // текущее число
    int last; // предыдущее число
    int count = 0; // число совпадений

    cin >> last;

    for (int i = 0; i < N - 1; ++i) {
        cin >> now;

        if (now == last) {
            count++;
        }

        last = now;
    }

    cout << count << endl;
    return 0;

}